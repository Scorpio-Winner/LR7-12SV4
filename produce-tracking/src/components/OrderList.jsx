import React from "react";
import Order from "./Order";

import classes from "../styles/OrderList.module.css";

const OrderList = ({ orders, orderStatuses, detailsHandler, deleteHandler, readonly}) => {
    if (orders.items.length === 0) {
        return <h3 className={classes.emptyList}>No orders found</h3>;
    }

    return (
        <div className={classes.orderList}>
            {orders.items.map((order) => (
                <Order
                    key={order.id}
                    orderData={order}
                    orderStatuses={orderStatuses}
                    detailsHandler={detailsHandler}
                    deleteHandler={deleteHandler}
                    readonly={readonly}
                />
            ))}
        </div>
    );
};

export default OrderList;
