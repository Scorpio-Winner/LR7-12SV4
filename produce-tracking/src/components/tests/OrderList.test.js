import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import OrderList from '../OrderList';  

describe('OrderList', () => {
  const mockOrders = {
    items: [
      {
        id: '1',
        title: 'Test Order',
        status: 'Pending',
        creation_date: '2022-02-02',
        customer: 'Test Customer',
        complete_time: '33 days',
        description: 'Test description'
      },
    ],
  };

  const mockOrderStatuses = ['Pending', 'In Progress', 'Completed'];

  it('renders correctly', () => {
    const { getByText } = render(
      <Router>
        <OrderList orders={mockOrders} orderStatuses={mockOrderStatuses} />
      </Router>
    );

    expect(getByText('Test Order')).toBeTruthy();
    expect(getByText('Pending')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <OrderList orders={mockOrders} orderStatuses={mockOrderStatuses} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
