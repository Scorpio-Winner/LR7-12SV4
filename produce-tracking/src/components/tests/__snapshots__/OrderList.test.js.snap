// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderList matches the snapshot 1`] = `
<div
  className="orderList"
>
  <div
    className="order"
  >
    <div>
      <div
        className="MuiBox-root css-q9c22"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-dtp3js-MuiTypography-root"
        >
          1
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-19li7pl-MuiTypography-root"
        >
          Test Order
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ytge06-MuiTypography-root"
        >
          Pending
        </p>
      </div>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1wgcmb5-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Details
      </button>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Create PDF
  </button>
</div>
`;
